### ISSUE TITLE
Will there be Maven packages available?

### ISSUE THREAD

**Issue by proninyaroslav** [frostwire/frostwire-jlibtorrent#295]
Hello. The last time Maven packages were updated was in 2018, do you plan to publish them again?

**Comment by gubatron** [comment]
yes, it's on the todo list after jlibtorrent 2.0 has been integrated to frostwire

**Comment by gubatron** [comment]
libtorrent4j is a fork of this library, it's slightly different.
they integrated with libtorrent 2 earlier than we did.

We recently finished integrating with the new API on the master branch, but we have yet to heavily test the new jars on FrostWire itself.

**Comment by wbpxre150** [comment]
Can you please do this. I am using this library in my app. And fdroid can't accept it with compiled libs. https://github.com/wbpxre150/unjumble/

**Comment by gubatron** [comment]
We still have not had a chance to integrate the jlibtorrent 2.0 library into FrostWire, until we do that we're not sure how well the 2.0 version is.

I understand your constraint, but I'm also curious to see how well it works, have you tried locally manually adding the jars to your classpath and libpath, does everything work for you? If so I'm more inclined to work on the maven distribution before integrating on frostwire.

**Comment by gubatron** [comment]
great news, this confirms my unit tests migrated well to 2.0.

I'll be coming back home from vacation next wednesday, I'll try to figure out maven publishing and put these out as soon as possible after that.

**Comment by gubatron** [comment]
I believe we're almost there, need to try these on FrostWire itself

[com.frostwire.jlibtorrent-1.2.19.0](https://github.com/frostwire/frostwire-jlibtorrent/packages/2586073?version=1.2.19.0)

[com.frostwire.jlibtorrent-windows-1.2.19.0](https://github.com/frostwire/frostwire-jlibtorrent/packages/2586077?version=1.2.19.0)


I did it using Github Packages, so you'll have to add github credentials into your environment before building to pull the dependencies, very weird that they ask for that to download dependencies...

```groovy
repositories {
    maven {
        url = uri("https://maven.pkg.github.com/frostwire/frostwire-jlibtorrent")
        credentials {
            username = project.findProperty("gpr.user") as String? ?: System.getenv("GITHUB_USERNAME")  // Replace with your GitHub username
            password = project.findProperty("gpr.key") as String? ?: System.getenv("GITHUB_PAT")  // Your PAT file token value with package:read permissions
        }
    }
}


dependencies {
    implementation 'com.frostwire:jlibtorrent:1.2.19.0'
    implementation 'com.frostwire:jlibtorrent-linux:1.2.19.0'
}
```

**Comment by wbpxre150** [comment]
Any chance you can put these on maven Central? Fdroid only trust certain maven repos.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by gubatron** [comment]
we lost access to our credentials there (sonatype), I'll see what I can do with, for this release let's keep it in our web server, we have a lot more flexibility.

It'd be awesome if we could configure the package at maven central to just mirror ours, maybe this is possible.
